src/shared/components/query/QueryStore.ts

```diff
@@ -297,8 +230,28 @@ export class QueryStore
 	{
 		let ids:string[] = this._selectedStudyIds.keys();
 		const selectableStudies = this.selectableStudiesSet;
-		ids = ids.filter(id=>!!selectableStudies[id]);
-		return this.forDownloadTab ? ids.slice(-1) : ids;
+		let selectableIds = ids.filter(id=>!!selectableStudies[id]);
+
+		if(ids.length !== selectableIds.length) {
+			let entities = this.caseIds.trim().split(/\s+/g);
+			let normStudies:string [] = entities
+			.map(entity=>{
+				let splitEntity = entity.split(':');

```

Comment:
since it seems we don't use the id portion, you could just make a regexp that matches the string prior to :.   This will then work for all strings, even those without : compound nature.